pkg/builder/build.go

```diff
@@ -218,3 +236,36 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	blder.ctrl, err = newController(name, blder.mgr, controller.Options{Reconciler: r})
 	return err
 }
+
+func (blder *Builder) doWebhook() error {
+	// Create a webhook for each type
+	gvk, err := apiutil.GVKForObject(blder.apiType, blder.mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	partialPath := strings.Replace(gvk.Group, ".", "-", -1) + "-" +
+		gvk.Version + "-" + strings.ToLower(gvk.Kind)
+
+	defaulter, isDefaulter := blder.apiType.(admission.Defaulter)
+	if isDefaulter {
+		mwh := admission.DefaultingWebhookFor(defaulter)
+		if mwh != nil {
+			path := "/mutate-" + partialPath
+			log.Info("registering a mutating webhook", "path", path)
+			blder.mgr.GetWebhookServer().Register(path, mwh)
+		}
+	}
+
+	validator, isValidator := blder.apiType.(admission.Validator)
+	if isValidator {
+		vwh := admission.ValidatingWebhookFor(validator)
```

Comment:
when we land conversion, we're going to want to automatically handle all registered versions of a given group-kind.  please add a comment.